src/test/java/io/jenkins/pluginhealth/scoring/service/PluginServiceIT.java

```diff
@@ -0,0 +1,57 @@
+package io.jenkins.pluginhealth.scoring.service;
+
+import static org.junit.Assert.assertEquals;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.repository.PluginRepository;
+
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@ContextConfiguration(initializers = PluginServiceIT.DockerPostgresDatasourceInitializer.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@Testcontainers
+public class PluginServiceIT {
+    @Autowired
+    private PluginRepository pluginRepository;
+    @Autowired
+    private PluginService pluginService;
+
+    @Container
+    private static final PostgreSQLContainer POSTGRE_SQL_CONTAINER = new PostgreSQLContainer("postgres:14.1")
+        .withDatabaseName("testdb")
+        .withUsername("sa")
+        .withPassword("sa");
+
+    public static class DockerPostgresDatasourceInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+        @Override
+        public void initialize(@NotNull ConfigurableApplicationContext applicationContext) {
+            TestPropertyValues.of(
+                "spring.datasource.url=" + POSTGRE_SQL_CONTAINER.getJdbcUrl(),
+                "spring.datasource.username=" + POSTGRE_SQL_CONTAINER.getUsername(),
+                "spring.datasource.password=" + POSTGRE_SQL_CONTAINER.getPassword()
+            ).applyTo(applicationContext);
+        }
+    }
+
+    @Test
+    public void shouldNotDuplicatePlugin() {
+        Plugin plugin = new Plugin("myPlugin", "https://github.com/jenkinsci/my-plugin", null);
+
+        pluginService.saveOrUpdate(plugin);
+        pluginService.saveOrUpdate(plugin);
+
+        assertEquals(1, pluginRepository.findAll().size());
```

Comment:
This test fails at this assertion with the error: `expected:<1> but was:<1878>`

Which is a little weird for me because I just saved one plugin object in the DB but somehow the test runs the full spring project (fetches all the data from the Update center and saves it into the DB) because only that's why there are 1877 (+1 object that we are adding as part of this test) plugins in the DB. Can you help me understand what am I missing?

Reply:
it is not weird. The initial import of the plugins is done on the main class of the application, which is ran when the test starts.

Reply:
I fixed that in https://github.com/jenkins-infra/plugin-health-scoring/pull/17/commits/f187fe3ca06b337b07e5fc42e02b9e0a6e935c63